aptos-move/block-executor/src/executor.rs

```diff
@@ -555,10 +674,21 @@ where
                     for (ap, write_op) in output.get_writes().into_iter() {
                         data_map.write(ap, write_op);
                     }
-                    // Calculating the accumulated gas of the committed txns.
-                    let txn_gas = output.gas_used();
-                    accumulated_gas += txn_gas;
-                    counters::SEQUENTIAL_PER_TXN_GAS.observe(txn_gas as f64);
+                    // Calculating the accumulated gas costs of the committed txns.
+                    let (total_gas, execution_gas, io_gas, storage_gas, storage_fee) =
+                        output.fee_statement();
+                    accumated_total_gas += total_gas;
+                    accumulated_execution_gas += execution_gas;
+                    accumulated_io_gas += io_gas;
+                    accumulated_storage_gas += storage_gas;
+                    accumulated_storage_fee += storage_fee;
```

Comment:
potentially use a FeeStatement (with ability to add / merge) struct?

Reply:
done